tests/test_validators.py

```diff
@@ -21,31 +21,47 @@
 }
 
 
-class SampleResource(object):
-    @jsonschema.validate(basic_schema)
-    def on_get(self, req, resp):
+class Resource(object):
+    @jsonschema.validate(schema_request=basic_schema)
+    def request_validated(self, req, resp):
         assert req.media is not None
+        return resp
+
+    @jsonschema.validate(schema_response=basic_schema)
+    def response_validated(self, req, resp):
+        assert req.media is not None
+        return resp
 
 
-class RequestStub(object):
+class GoodData(object):
     media = {'message': 'something'}
 
 
+class BadData(object):
+    media = {}
+
+
 @skip_py26
-def test_jsonschema_validation_success():
-    req = RequestStub()
+def test_jsonschema_request_validation_success():
+    data = GoodData()
+    assert Resource().request_validated(GoodData(), data) is data
 
-    res = SampleResource()
-    assert res.on_get(req, None) is None
+
+@skip_py26
+def test_jsonschema_request_validation_failure():
+    with pytest.raises(falcon.HTTPBadRequest) as err:
+        Resource().request_validated(BadData(), None)
+        assert err.value.description == '\'message\' is a required property'
 
 
 @skip_py26
-def test_jsonschema_validation_failure():
-    req = RequestStub()
-    req.media = {}
+def test_jsonschema_response_validation_success():
+    data = GoodData()
+    assert Resource().response_validated(GoodData(), data) is data
 
-    res = SampleResource()
-    with pytest.raises(falcon.HTTPBadRequest) as err:
-        res.on_get(req, None)
 
-    assert err.value.description == '\'message\' is a required property'
+@skip_py26
```

Comment:
Once you rebase on master, you can remove these `@skip_py26` decorators, as we have removed 2.6 support.